### ISSUE TITLE
When deleting a provider with associated Migration Plan or Infrastructure mapping, the plan and mappings should also be deleted

### ISSUE THREAD

**Issue by kedark3** [ManageIQ/miq_v2v_ui_plugin#484]
https://bugzilla.redhat.com/show_bug.cgi?id=1596377
If we do not delete infra mappings, as shown in the screenshot below, we see a mapping with 0 Clusters/Networks/Datastores.
![screenshot from 2018-07-11 11-55-16](https://user-images.githubusercontent.com/8180469/42584501-4fd6e8c6-8501-11e8-90d3-b84164821b38.png)

**update 8/7**: the infrastructure mappings /provider deletion is handled by this fix:
https://github.com/ManageIQ/miq_v2v_ui_plugin/pull/503 

In the future, we may also to decide to handle plan/provider deletion showing warning messages as done in this [PR](https://github.com/ManageIQ/miq_v2v_ui_plugin/pull/439) ("Infrastructure Mapping does not exist" vs "Provider does not exist"). Currently, the API to do this is quite limited, so I suggest holding this for now.

**Comment by priley86** [comment]
It’s fine if you want to place the issue here, but I don’t think is a UI change in v2v repo.

**Comment by kedark3** [comment]
@bthurber I think I placed it in wrong repo?

**Comment by kedark3** [comment]
with 5.9.4.2 I can see that removing provider does not cause migration page to go blank. I can see "Red Cross" for invalid mappings.

**Comment by priley86** [comment]
@kedark3 @vconzola - this is not as simple as I initially thought for Plans. In the case of Infra Mappings, we are already doing the mapping logic to visualize cluster/provider associations. However we do not currently do anything similar for Plans. To make this possible, I would need to add the logic to go down the following bunny trail:

```
Plan (ServiceTemplate)--> Associated Mappings --> Clusters for Associated Mappings --> Provider(s) for Clusters)
```

Today we do not do that sort of thing on the Overview for a Plan, but it is possible by grabbing the `ext_management_system.id` (plan id) associated with Clusters (since we have them in memory), and going down that trail...

Long story short - I think this is best handled on the backend to avoid more UI pollution in that case. @kedark3 - would you mind just testing this scenario for an orphaned plan (with a deleted plan) and let us know if you get a runtime error, or if the migration fails gracefully? I think it is acceptable behavior and this is definitely an edge case / operator error.

**Comment by kedark3** [comment]
@priley86 I can test this, but what do you mean by 

>  runtime error, or if the migration fails gracefully?

**Comment by priley86** [comment]
does the CF application actually break or no longer respond? Or does the UI just show error messages during the migration? If it fails gracefully and the UI is still responsive, I don't see this as hugely pressing. I am not able to test it here to see the actual behavior in this scenario.

**Comment by kedark3** [comment]
Yes, UI is still responsive.

**Comment by priley86** [comment]
@kedark3 thanks. This is good to know. Is there anything in the console? What specific error do you see in the UI log? It would be nice to just add anything else we can in the documentation about this that might help the user.

@vconzola I think we can close this issue after we confirm that this is added to the documentation. The operator should not remove providers/clusters/networks/storages associated with the migration they are attempting to run. If they do so, the UI will remain responsive and indicate any missing cluster/network/storages for the infrastructure mapping (noted in #503). If the migration plan is attempted, it will fail gracefully.

**Comment by kedark3** [comment]
@priley86 I did not see any related errors in the browsers console. So I think we are good.

**Comment by kedark3** [comment]
@priley86  can we at least hide create migration plan button if not enough providers exists?

**Comment by vconzola** [comment]
@kedark3 We don't hide things in the UI. That's poor UX. The user has to take some ownership of their environment and understand that if they don't have providers in CF they can't do migration. This will be documented. We could, of course, check for this unlikely scenario (and many others), but as @priley86 explained the logic required to do so is complex and would add to our existing "UI pollution". If you think this scenario is common enough to be a concern in the wild, it should be addressed in the backend code, not the UI.
